src/client.ts

```diff
@@ -909,13 +918,27 @@ export class Client<Ctx extends unknown = null> {
     /**
      * Abrupt socket closures should report failed
      */
-    ws.onclose = () => {
+    ws.onclose = (event: CloseEvent | Event) => {
       if (!onFailed) {
         this.onUnrecoverableError(new Error('Got websocket closure but no `onFailed` cb'));
 
         return;
       }
 
+      if (WebSocketClass === EIOCompat) {
+        if (!didReceiveAnyCommand) {
+          // The polling implementation doesn't convey the Websocket close
+          // event. Let's assume that we need to request a new token.
+          this.connectionMetadata = null;
+        }
+      } else if (Object.prototype.hasOwnProperty.call(event, 'code')) {
```

Comment:
there should be an easier way to do this. think this should work:
```suggestion
      } else if ('code' in event) {
```